src/auth0/handlers/actions.js

```diff
@@ -96,21 +65,21 @@ function mapSecrets(secrets) {
   }
 }
 
-function mapCurrentVersion(currentVersion) {
-  if (currentVersion) {
-    return ({ ...currentVersion, secrets: mapSecrets(currentVersion.secrets) });
+function mapDeployedVersion(deployedVersion) {
+  if (deployedVersion) {
+    return ({ ...deployedVersion, secrets: mapSecrets(deployedVersion.secrets) });
   }
 }
 
-function mapAction(action, version) {
+function mapAction(action, deployedVersion) {
   return {
     ...action,
-    code: version ? version.code : action.code,
-    deployed: !!version,
-    secrets: version ? mapSecrets(version.secrets) : mapSecrets(action.secrets),
-    dependencies: version ? version.dependencies : action.dependencies,
-    status: version ? version.status : action.status,
-    current_version: mapCurrentVersion(version)
+    code: deployedVersion ? deployedVersion.code : action.code,
+    deployed: !!deployedVersion,
```

Comment:
There are many properties mapped into the returned value via line 76 above (`...action,`).

Is this `mapAction()` function intended to fill missing properties in `deployedVersion` with data from `action`?

But this function appears to taking all data from `action` when `deployedVersion` is undefined?

Reply:
Hi @shushen , this function intends to override some `action` fields if you have or not a `deployed_version`, because for `deploy-cli` we decide not to repeat version fields like `code`, `dependencies`, `secrets`, etc (fields from the `deployed_version` and the `draft` version), if the `deployed_version` exists, the top-level `action` fields will be filled with `deployed_version` data and `deployed` field will be true, if it not exists, we keep the `action` data and `deployed` field will be false.

This is an example of an exported action with `deployed_version`

![image](https://user-images.githubusercontent.com/60328073/117145836-f788e200-ad89-11eb-9664-744576ba9a20.png)